api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupTxRTT2019Report.java

```diff
@@ -185,22 +186,33 @@ public void addIndicator(CohortIndicatorDataSetDefinition dsd, String key, Strin
         dsd.addColumn(key, label, Mapped.mapStraightThrough(ci), dimensionOptions);
     }
 
-    public CohortDefinition getPatientsWithNoClinicalContactsFor28DaysAndAbove() {
+    public CohortDefinition getPatientsWithNoClinicalContactsForAbove28DaysByBeginningOfPeriod() {
         String query = "select person_id from (select o.person_id,last_enc_date,max(o.value_datetime)next_visit from obs o left join \n" +
                 "(select patient_id,max(encounter_datetime)last_enc_date from encounter where encounter_datetime <=:startDate group by patient_id) last_encounter \n" +
                 "on o.person_id=patient_id where o.concept_id=5096 and o.value_datetime <= :startDate group by person_id)t1 \n " +
-                "where next_visit > last_enc_date and datediff(:startDate,next_visit)>=28 ";
+                "where next_visit > last_enc_date and datediff(:startDate,next_visit)> 28 ";
         SqlCohortDefinition cd = new SqlCohortDefinition(query);
         cd.addParameter(new Parameter("startDate", "startDate", Date.class));
         return  cd;
     }
 
+    public CohortDefinition getLostPatientsDuringPeriodAndReturnToCareWithinSamePeriod() {
```

Comment:
@dbaluku cant this cohort be generalised for use in other coming reports, looks like its a cohort we can use.

Reply:
@solemabrothers  this cohort is too specific for RTT it cant be used anywhere